### ISSUE TITLE
policyInNv used uninitialized in TPMT_HA_Unmarshal()?

### ISSUE THREAD

**Issue by setharnold** [stefanberger/libtpms#311]
https://github.com/stefanberger/libtpms/blob/8590849b65260d71d22e5509f0c3ba44079cf780/src/tpm2/EACommands.c#L1144

Hello, this may be related to https://github.com/stefanberger/libtpms/issues/310 -- perhaps there's an idiom or similar here I'm misunderstanding.

It looks to me like `TPMT_HA_Unmarshal()` calls into `TPMI_ALG_HASH_Unmarshal()`, which appears to try to save and restore the stack garbage that was in `policyInNv` in case an initialization fails?

What's the point of saving aside and restoring the garbage, assuming I've understood this correctly?

Thanks

**Comment by stefanberger** [comment]
Oh well, pushed patches to master. Fixed :-)

**Comment by setharnold** [comment]
Okay, cool; I'll skip the other instances of similar-looking things in my last few minutes of going through Coverity output. ;) Thanks Stefan!

**Comment by stefanberger** [comment]
With latest master you see 7 error reports from Coverity, right?

**Comment by setharnold** [comment]
I'm actually working through `0.9.0-0ubuntu4` from the Ubuntu Jammy development release, so it might be different than your master branch. (Some of the findings have already been fixed; I try to take account of this when reporting back upstream the things that seem like real positives.)

The Coverity report we generated on this package has 26 items. I'd be happy to send along the textual output if you'd find it helpful for something.

Thanks
